compiler/passes/checkResolved.cpp

```diff
@@ -56,9 +57,24 @@ checkConstLoops() {
   }
 }
 
+static void checkForClassAssignOps(FnSymbol* fn) {
+  if (fn->getModule()->modTag == MOD_USER) {
+    if (strcmp(fn->name, "=") == 0 &&
+        fn->formals.head) {
+      ArgSymbol* formal = toArgSymbol(toDefExpr(fn->formals.head)->sym);
+      Type* formalType = formal->type->getValType();
+      if (isOwnedOrSharedOrBorrowed(formalType) ||
+          isUnmanagedClass(formalType)) {
+        USR_FATAL_CONT(fn, "Can't overload assignments for class types");
```

Comment:
@mppf - which flavor of `isClass()` can we use instead of `isOwnedOrSharedOrBorrowed() || isUnmanagedClass()` above?

Reply:
@mppf: Curious about Vass's question:  Is there a better practice here?

Reply:
`isClassLikeOrManaged` is what I would use here, personally.

Reply:
I think we should work on improving the naming of these routines.  To me, the name "isClassLike" suggested that it accepted cases other than classes, which I wasn't interested in considering here, so I stayed away from those variants not knowing what they would let through.  Similarly, it seemed odd that isClass() would return `true` for things that aren't classes in the language's view.

Reply:
Yep. It's the way it is for historical reasons, not because they make the most sense.

Reply:
OK.  Are there (known) cases where isClassLikeOrManaged will give a different result than what I've got here (is owned or shared or borrowed or unmanaged)?  I'm feeling slightly hesitant to change since it feels like almost a wash in code clarity and this has already passed testing.

Reply:
It would be fine with me to leave it as-is. I don't know of any case that would be different.